etc/tune_it.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/sh
+
+# A linear congruential generator is sufficient for the purpose.
+SEED=3735928559
+LCG() {
+  SEED=$(((1103515245 * $SEED + 12345) % 2147483648))
+  echo $SEED
+}
+
+MPWD=$(pwd)
+FILE_NAME="tuning_list"
+BNCORE_C="../bncore.c"
+BACKUP_SUFFIX=".orig"
+RNUM=0;
+#############################################################################
+# It would be a good idea to isolate these processes (with e.g.: cpuset)    #
+#                                                                           #
+# It is not a good idea to e.g: watch high resolution videos while this     #
+# test are running if you do not have enough memory to avoid page faults.   #
+#############################################################################
+
+# Number of rounds overall.
+LIMIT=100
+# Number of loops for each input.
+RLOOPS=10
+# Offset (> 0) . Runs tests with asymmetric input of the form 1:OFFSET
+OFFSET=1
+# Number of positive results (TC-is-faster) accumulated until it is accepted
+# (mainly for "noisy" environments, especially if isolation is not possible)
+LAG=1
+# Keep the temporary file $FILE_NAME. Set to 0 (zero) to remove it at the end.
+KEEP_TEMP=1
+
+echo "You might like to watch the numbers go up to $LIMIT but it will take a long time!"
+
+echo "km ks tc3m tc3s" > $FILE_NAME
+i=1
+while [ $i -le $LIMIT ]; do
+   RNUM=$(LCG)
+   echo $i
+   "$MPWD"/tune -t -r $RLOOPS -L $LAG -S "$RNUM" -o $OFFSET >> $FILE_NAME
+   i=$((i + 1))
+done
+
+i=1
+SUMMARY=""
+TMP=""
+while [ $i -le 4 ]; do
+   if [ $i -eq 1 ]; then
+      v="v"
+   else
+      v=""
+   fi
+   TMP=$(cat  $FILE_NAME | cut -d' ' -f$i | "$MPWD"/statistic_summary_single_column $v;echo z)
+   SUMMARY="$SUMMARY${TMP%z}"
+   i=$((i + 1))
+done
+printf '%s\n' "$SUMMARY"
+if [ $KEEP_TEMP -eq 0 ]; then
+   rm -v $FILE_NAME
+fi
+
+echo "Writing cut-off values to \"bncore.c\" using the median."
+echo "In case of failure: a copy of \"bncore.c\" is in \"bncore.c.orig\""
+
+cp -v $BNCORE_C $BNCORE_C$BACKUP_SUFFIX
+
+cat << END_OF_INPUT > $BNCORE_C
+#include "tommath_private.h"
+#ifdef BNCORE_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+
+/* Known optimal configurations
+
+ CPU                    /Compiler     /MUL CUTOFF/SQR CUTOFF
+-------------------------------------------------------------
+ Intel P4 Northwood     /GCC v3.4.1   /        88/       128/LTM 0.32 ;-)
+ AMD Athlon64           /GCC v3.4.4   /        80/       120/LTM 0.35
+
+*/
```

Comment:
Maybe remove those old comments?

Reply:
Can do. Shall I?

Reply:
Yes, they don't make sense if the values are updated?

Reply:
Not even for historic reasons?
No?
OK.